testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -197,6 +197,45 @@ def test_start_stop_step(u, run_kwargs, frames):
     assert_almost_equal(an.times, frames+1, decimal=4, err_msg=TIMES_ERR)
 
 
+@pytest.mark.parametrize('run_kwargs, frames', [
+    ({'frames': [4, 5, 6, 7, 8, 9]}, np.arange(4, 10)),
+    ({'frames': [0, 2, 4, 6, 8]}, np.arange(0, 10, 2)),
+    ({'frames': [4, 6, 8]}, np.arange(4, 10, 2)),
+    ({'frames': [True, True, False, True, False, True, True, False, True,
```

Comment:
Also add `[0, 3, 4, 3, 5]` with a repeated frame.

Otherwise I like that you're checking `an.frames` !

Reply:
Is the proper behavior that it will run frame 3 twice?

Reply:
yes